subsys/testsuite/coverage/coverage.c

```diff
@@ -237,21 +265,22 @@ void gcov_coverage_dump(void)
 	printk("\nGCOV_COVERAGE_DUMP_START");
 	while (gcov_list) {
 
+		dump_on_console_start(gcov_list->filename);
```

Comment:
should this call be moved to line 282 to avoid output in case of the following errors ? or this is intentional to have the header always printed ?

Reply:
Good point, I did not notice that I changed this behavior.
However, if an error occurs, no more data are printed (from other files). An error about buffer allocation is sent to the console, so the output will not be complete and will have something else than the correct sequence (i.e. error message) anyway.
Leaving it as it is will allow us to see which file's data were too big to be printed - that was missing previously (that could be added to the error message).

Twister output parser will handle correctly that changed behavior i.e. still there will not be complete coverage dumped.

If you suggest changing it to the original sequence, I am okay with that.
However, I see the benefits of leaving it as it is - more information about the context where the error occurs (and no particular downside except more information printed)